### ISSUE TITLE
Xcode 8 beta 5 + iOS10 Sim: waitForViewWithAccessibilityLabel: fails

### ISSUE THREAD

**Issue by pantuspavel** [kif-framework/KIF#894]
Getting error:
`The step timed out after 10.00 seconds: Failed to find accessibility element with the label "MyLabel"` on Xcode 8 beta 5 + iOS10 Sim setup. (`waitForViewWithAccessibilityLabel:` method)

Xcode 8 beta 5 + iOS9 Sim setup works fine.

Please let me know if any additional information needed. Will gladly update the issue.

**Comment by phatmann** [comment]
Are you using KIF HEAD branch?

**Comment by pantuspavel** [comment]
@phatmann 
`pod 'KIF', '3.4.2'`

**Comment by phatmann** [comment]
Please try `pod 'KIF', :git => https://github.com/kif-framework/KIF.git` to use HEAD branch. You should not expect a release branch to necessarily be compatible with a version of iOS in beta.

**Comment by pantuspavel** [comment]
Thanks for suggestion.
I just tried it and unfortunately it didn't help.
Same situation - works on iOS9 sim, doesn't on iOS10 sim.

**Comment by phatmann** [comment]
Then we could use your help debugging this! :-)

**Comment by pantuspavel** [comment]
Absolutely! 👍 
Any suggestions where I should look at or pay attention to?

**Comment by phatmann** [comment]
If I could tell you that, I could fix the bug already ;-)

**Comment by pantuspavel** [comment]
Fair! 😄 

So I was able to track down the problem.
In iOS 9 and older it was possible to do `waitForViewWithAccessibilityLabel:` for UILabels without actually setting `accessibilityLabel`.
```
self.textLabel.text = item.text;
```
In iOS10 it's not enough, so I just did the following and it worked for me:
```
self.textLabel.text = item.text;
self.textLabel.accessibilityLabel = item.text;
```

**Comment by phatmann** [comment]
Thanks for the follow up. This is an important change to know about in IOS 10. I am closing this issue, since KIF is correctly looking for an accessibilityLabel. If you or someone thinks that `waitForViewWithAccessibilityLabel` should find labels without an `accessibilityLabel`, please file a PR with this change.

**Comment by knellr** [comment]
(this issue was closed while I was typing this but I'm posting anyway..)

It looks like the accessibility label is nil in beta 5 on e.g. text labels, but the value is set. My tests pass if I modify the following:
```
accessibilityElement:view:withLabel:value:traits:fromRootView:tappable:error:
```
to swap label and value if no element is found:
```
UIAccessibilityElement *element = [self accessibilityElementWithLabel:label value:value traits:traits fromRootView:fromView error:error];
    if (!element) {
        if (label != nil && value == nil) {
            return [self accessibilityElement:foundElement view:foundView withLabel:nil value:label traits:traits fromRootView:fromView tappable:mustBeTappable error:error];
        } else {
            return NO;
        }
    }
```

Assuming this isn't a bug, it might be best to require users to be explicit about whether they want the value or the label - though that would break a lot of tests 💩 

If we're being explicit it would be useful to have `waitForViewWithValue:` etc for convenience.

**Comment by phatmann** [comment]
I recommend that you use `KIFUIViewTestActor`. It allows you to use different predicates, and one of them is `usingValue`.

**Comment by bnickel** [comment]
This sounds like a bug in the latest beta (or the loading of the accessibility framework).  I doubt Apple would intentionally remove automatic accessibility labels for UILabel since it would cripple accessibility across the platform.

**Comment by justinseanmartin** [comment]
Completely agree. I'm going to file a radar just to be safe though, and would recommend others do the same.

I assume this isn't a simulator only bug? And also affects their UI Testing framework? It would be less likely to see traction if those two continue to work.

**Comment by knellr** [comment]
It does seem like a bug, but UITest and voiceover are unaffected, so this could also be a deliberate choice. Will open a radar.
